remote/src/app/modals/app-remote-dark-mode-switch/app-remote-dark-mode-switch.tsx

```diff
@@ -0,0 +1,68 @@
+import { Component, h, Listen, State } from '@stencil/core';
+import { set, get } from 'idb-keyval';
+
+@Component({
+    tag: 'dark-mode-switch',
+})
+export class DarkModeSwitch {
+
+    @State()
+    private darkModePreference: Boolean = false;
+    private domBodyClassList: DOMTokenList = document.body.classList;
+
+    // error handling if the idb-keyval throws an error
+    private async idbKeyValPromiseHandler (promise): Promise<any> {
+        return Promise.resolve(promise)
+        .then( valueOfPromise => valueOfPromise)
+        .catch(err => Error(err));
+    }
+
+    private resolveDarkModePreference(): Boolean | Promise<any> {
+        let getDarkModePreferenceFromIdb: Promise<any> = this.idbKeyValPromiseHandler(get('darkModePreferences'));
+        let getDarkModePreferenceFromMedia = window.matchMedia('(prefers-color-scheme: dark)');
+        if( getDarkModePreferenceFromIdb instanceof Error ){
+            return false;
+        }
+        let isDarkModePreferenceDetermined: Boolean | Promise<any> = 
+        getDarkModePreferenceFromIdb || 
+        getDarkModePreferenceFromMedia.matches;
```

Comment:
@peterpeterparker 👀 so we could do something like this where we check for the dark mode preference when the application is loaded. So here we are checking the media query and the key val stuff. What do you think? If you wanted to added a full listener I could add a @watch event for `onchange`? Im not super well versed in stencil, but I think that could work?

Reply:
Yes exactly we could do something like this. The remote UX will be refactored, so based on that following implementation we will be able to fine tune the solution.

Just note, currently, `this.darkModePreference = await this.resolveDarkModePreference();` resolve `undefined` in my case which doesn't seems correct. Maybe a promise not resolving?